### PR TITLE
Default empty array for previously flattened lists and maps in query compatible services

### LINKED ISSUE #2947: aws-sdk-sqs raises NoMethodError when messages not found
### Describe the bug

Error occaisonally happens inside `QueuePoller#poll`.

The error message indicates `messages` is nil.

### Expected Behavior

No error happens.

### Current Behavior

```
 undefined method `empty?' for nil:NilClass
  /opt/rubies/ruby-2.7.8/lib/ruby/gems/2.7.0/gems/aws-sdk-sqs-1.67.0/lib/aws-sdk-sqs/queue_poller.rb:358:in `block (2 levels) in poll'
```

### Reproduction Steps

Needs more research

### Possible Solution

_No response_

### Additional Information/Context

The error starts on Nov 18, so AWS side might have changed at the timing.

### Gem name ('aws-sdk', 'aws-sdk-resources' or service gems like 'aws-sdk-s3') and its version

aws-sdk-sqs-1.67.0

### Environment details (Version of Ruby, OS environment)

ruby-2.7.8 on Amazon Linux 2

### PR DESCRIPTION
**Pull request by alextwoods**
Fixes #2947 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

1. To make sure we include your contribution in the release notes, please make sure to add description entry for your changes in the "unreleased changes" section of the `CHANGELOG.md` file (at corresponding gem). For the description entry, please make sure it lives in one line and starts with `Feature` or `Issue` in the correct format.

2. For generated code changes, please checkout below instructions first:
  https://github.com/aws/aws-sdk-ruby/blob/version-3/CONTRIBUTING.md

Thank you for your contribution!

### CODE REVIEW

File: gems/aws-sdk-sqs/lib/aws-sdk-sqs/queue_poller.rb [hunk 1]
```diff
@@ -406,7 +406,7 @@ def delete_messages(messages)
 
       def get_messages(config, stats)
         config.before_request&.call(stats)
-        messages = send_request(config).messages
+        messages = send_request(config).messages || []
```
Comment (mullermp) [inline gems/aws-sdk-sqs/lib/aws-sdk-sqs/queue_poller.rb]: I understand this as a temp fix, but shouldn't messages be parsed as an empty array already?

Comment (alextwoods) [reply gems/aws-sdk-sqs/lib/aws-sdk-sqs/queue_poller.rb]: I don't think so - I think in aws json nil array members should be desrialized as nil and not default to `[]`.

Comment (mullermp) [reply gems/aws-sdk-sqs/lib/aws-sdk-sqs/queue_poller.rb]: Then this is a breaking change to the service - `receive_message` API has a different return value, and breaks other usages.

Comment (mullermp) [reply gems/aws-sdk-sqs/lib/aws-sdk-sqs/queue_poller.rb]: I added a plugin as a temporary solution? But I think when we see query compatible trait and also a list, we need to default it to an empty list.

**Review by alextwoods — COMMENTED** [review]
Lgtm

**Review by jterapin — APPROVED** [review]

**Review by kstich — APPROVED** [review]

File: gems/aws-sdk-sqs/lib/aws-sdk-sqs/queue_poller.rb (hunk 1 above)
Comment (geeksam) [reply gems/aws-sdk-sqs/lib/aws-sdk-sqs/queue_poller.rb]: I see that this change was backed out, but nevertheless wish to note for future reference that, if one's style guide / cyborg law enforcement officers permit, the `Kernel#Array` method is a nice way to guarantee that one has an Array :)

**Pull request merged**